presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayUnionFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.operator.aggregation.TypedSet;
+import com.facebook.presto.operator.scalar.annotations.ScalarFunction;
+import com.facebook.presto.operator.scalar.annotations.TypeParameter;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.SqlType;
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+import it.unimi.dsi.fastutil.longs.LongSet;
+
+import javax.annotation.Nullable;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+@ScalarFunction("array_union")
+@Description("Union elements of the two given arrays")
+public final class ArrayUnionFunction
+{
+    private ArrayUnionFunction() {}
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public static Block union(
+            @TypeParameter("E") Type type,
+            @Nullable @SqlType("array(E)") Block leftArray,
```

Comment:
@cberner Thanks for your replay. I just think `NULL` represents an unknown value or nothing, when a not NULL array union with `NULL`， and it should be return not NULL array but `NULL`. This is different with `array_intersect`. And if `array_union` input is two `NULL`, it will return `NULL`. 
If we don't do this, to make sure to get right result, we will need use like:
```
array_union(if(arr1 is null, ARRAY[], arr1), if(arr2 is null, ARRAY[], arr2))
```
This is so complex.